snuba/utils/streams/batching.py

```diff
@@ -251,5 +267,5 @@ def _flush(self, force: bool = False) -> None:
         self._reset_batch()
 
     def _commit(self) -> None:
-        offsets = self.consumer.commit()
+        offsets = self.__commit_retry_policy.call(self.consumer.commit)
```

Comment:
why not putting this at the abstract level ? Now that we can provide the retry policy in the constructor, I don't see any reason to support this only in the batching consumer.
Specifically, since this level adds batching to a consumer provided through the constructor, is there any reason not to provide the retry policy to the consumer that is passed to the constructor and rely on that one to perform the retry ?

Reply:
I think moving this to the consumer rather than the batching consumer makes sense — it also seems to echo this comment a bit: https://github.com/getsentry/snuba/pull/546#discussion_r340863837

I'll start that process here.